### PR TITLE
feat: pokemon details action

### PR DESCRIPTION
**Pull request by Miqqy0069**
-tried before and after for GetPokemonDetails Action
-added reduce for pokemonDetails

Before 
image_720

Reduce
image_720

After 
image_720

Analyze 
image_360

Dart Format 
image_720

Working App
image_480

### CODE REVIEW

File: lib/state/app_state.dart [hunk 1]
```diff
@@ -10,6 +11,8 @@ class AppState with _$AppState {
   factory AppState({
     @Default('') String sample,
     @Default(<Pokemon>[]) List<Pokemon> pokemons,
+    @Default(PokemonDetails(height: 0, weight: 0, baseExperience: 0, abilities: [], moves: [], stats: []))
```
Comment (shawn-ffufm) [inline lib/state/app_state.dart]: make this nullable instead with null as its default value 🙂

**Review by shawn-ffufm — CHANGES_REQUESTED** [review]
@Miqqy0069 see minor feedback 🙂

File: lib/state/app_state.dart (hunk 1 above)
Comment (Miqqy0069) [reply lib/state/app_state.dart]: Resolved po sir

File: lib/state/action/pokemon_actions.dart [hunk 2]
```diff
@@ -13,3 +13,17 @@ class GetPokemonsAction extends ReduxAction<AppState> {
     return state.copyWith(pokemons: pokemons);
   }
 }
+
+/// Getting the pokemon Details from pokemon api
+class GetPokemonDetailsAction extends ReduxAction<AppState> {
+  final String pokemonName;
```
Comment (shawn-ffufm) [inline lib/state/action/pokemon_actions.dart]: move this below of your constructor 🙂

**Review by shawn-ffufm — CHANGES_REQUESTED** [review]
see minor feedback 🙂

File: lib/state/action/pokemon_actions.dart (hunk 2 above)
Comment (Miqqy0069) [reply lib/state/action/pokemon_actions.dart]: Resolved po

File: lib/features/home_page.dart [hunk 3]
```diff
@@ -8,6 +8,7 @@ class HomePage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     StoreProvider.dispatch(context, GetPokemonsAction());
```
Comment (shawn-ffufm) [inline lib/features/home_page.dart:12]: add a todo comment that these will be removed later 🙂

File: lib/features/home_page.dart [hunk 4]
```diff
@@ -8,6 +8,7 @@ class HomePage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     StoreProvider.dispatch(context, GetPokemonsAction());
+    StoreProvider.dispatch(context, GetPokemonDetailsAction(pokemonName: 'ditto'));
```
Comment (shawn-ffufm) [inline lib/features/home_page.dart]: kindly place the ditto string in a constant and have a todo comment as well 🙂

**Review by shawn-ffufm — CHANGES_REQUESTED** [review]
see pr feedback

File: lib/features/home_page.dart (hunk 3 above)
Comment (Miqqy0069) [reply lib/features/home_page.dart:12]: resolved

File: lib/features/home_page.dart (hunk 4 above)
Comment (Miqqy0069) [reply lib/features/home_page.dart]: resolved

File: lib/utilities/string_constants.dart [hunk 5]
```diff
@@ -3,3 +3,5 @@ const homePageText = 'Home Page';
 
 const offset = '0';
 const limit = '151';
+
+const pokemonDitto = 'ditto';
```
Comment (shawn-ffufm) [inline lib/utilities/string_constants.dart:8]: you forgot the todo comment for this 😛

**Review by shawn-ffufm — CHANGES_REQUESTED** [review]
see minor feedback

File: lib/utilities/string_constants.dart (hunk 5 above)
Comment (Miqqy0069) [reply lib/utilities/string_constants.dart:8]: resolved

**Review by shawn-ffufm — APPROVED** [review]

**Pull request merged**